### PR TITLE
Add on DiskTree: an on-disk hextree

### PR DESCRIPTION
**Pull request by JayKickliter**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,19 +19,27 @@ Location to value mapping.
...
+disktree = [
+  "byteorder",
+  "memmap",
+  "serde",
+]
 serde = ["dep:serde"]
 
 [dependencies]
+byteorder = { version = "1", optional = true }
+leb128 = { version = "0.2.5", optional = true }
```
Comment (JayKickliter) [inline Cargo.toml:32]: not used

File: src/disktree/dptr.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    pub(crate) const fn null() -> Dptr {
+        Dptr(Self::NULL)
+    }
+
+    pub(crate) const fn size() -> u64 {
+        Self::DISK_REPR_SZ as u64
+    }
+
+    /// Read 5 bytes from disk and parses them as litte-endient `u64`.
```
Comment (Vagabond) [inline src/disktree/dptr.rs:30]:
```suggestion
    /// Read 5 bytes from disk and parses them as little-endian `u64`.
```

File: src/disktree/dptr.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        R: Read,
+    {
+        let mut buf = [0u8; size_of::<u64>()];
+        src.read_exact(&mut buf[..Self::DISK_REPR_SZ])?;
+        let dptr = u64::from_le_bytes(buf);
+        Ok(dptr.into())
+    }
+
+    /// Read 5 * `n` bytes from disk, for up to n=7, and parses them as
+    /// litte-endien `u64`s.
```
Comment (Vagabond) [inline src/disktree/dptr.rs:42]:
```suggestion
    /// little-endian `u64`s.
```

File: src/disktree/iter.rs [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    ///
+    /// See  [`Iter::node_buf`].
+    fn recycle_node_buf(&mut self, buf: Vec<(u8, Dptr)>) {
+        debug_assert!(buf.is_empty());
+        self.recycle_bin.push(buf);
+    }
+
+    // We've encountered an IO error. We're still going to return
+    // `Some` with the contents of the user's deserializer, but let's
+    // make sure we never yeild another value by clearing stack.
```
Comment (Vagabond) [inline src/disktree/iter.rs:89]:
```suggestion
    // make sure we never yield another value by clearing stack.
```

File: src/disktree/iter.rs [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    fn stop_yeilding(&mut self) {
```
Comment (Vagabond) [inline src/disktree/iter.rs:90]:
```suggestion
    fn stop_yielding(&mut self) {
```

**Review by Vagabond — APPROVED** [review]
Nothing looks too wacky to me.

**Pull request merged**